src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,105 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("get method called");
+    response.setContentType("text/html");
+    ArrayList<String> result = new ArrayList<String>();
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      //check if this user has registered
+      String urlToRedirectToAfterUserLogsOut = "/index.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      result.add(userEmail);
+      result.add(logoutUrl);
+      if (hasRegistered(userEmail)) {
+        // return third element if user already registered
+        String userType = getUserType(userEmail);
+        result.add(userType);
+      }
+    } else { // not yet signed in
+      String urlToRedirectToAfterUserLogsIn = "/signin.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      //somewhere over here need to know if this is for charity / restaurant
+      // how to know?
```

Comment:
My idea was that when a user logs in, make a GET request to `user` entity and get `user_type`. Based on `user_type`, invoke relevant front end features. Hope that helps

Reply:
Actually, now that I think of it why do we need to know the `user_type` at this point?

Reply:
I have resolved this issue, thanks